auth_mellon_handler.c

```diff
@@ -130,7 +123,7 @@ static char *am_generate_metadata(apr_pool_t *p, request_rec *r)
 
     sp_entity_id = cfg->sp_entity_id ? cfg->sp_entity_id : url;
 
-    if (cfg->sp_cert_file) {
+    if (cfg->sp_cert_file && cfg->sp_cert_file->contents) {
```

Comment:
You are correct, however the NULL pointer check seems like an inexpensive protection against any future code change which might allow mellon to initialize if the contents could not be read at start-up. Note that other configuration files do not have this same requirement to read the contents and instead postpone their validation in a lazy fashion on first use, .e.g when loaded into Lasso. It would not be a stretch to imagine unifying all file reads to be lazy on first use. If that change were introduced it would also be easy to miss modifying this check and hence introducing an unintended segfault. It seems worthwhile to guard against a segfault especially when the cost is so low.

Reply:
Just to be clear, I'm not really concerned about the CPU cycles of the extra NULL-check. When I ask questions it is mainly because there is something that I don't understand, and in this case it was why there was extra checks here, but not in many other locations where `somefile->contents` was accessed. However, upon further inspection, I see that the other places allow NULL-arguments to be used, so it isn't really a problem there.

(Btw., there is a reason why these options are read at startup -- these private keys & certificates can (and should) be locked down so that only root can read them. Once Apache starts it will load them, and then switch to its non-privileged user account, so it cannot load the files after startup.)